syslog/syslog.go

```diff
@@ -109,6 +109,11 @@ func Dial(clientHostname, network, raddr string, rootCAs *x509.CertPool, connect
 	// dial once, just to make sure the network is working
 	conn, err := dial(network, raddr, rootCAs, connectTimeout)
 
+	// do not spawn a watch go routine if we cannot connect
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I believe that it's intended to not return earlier here so the connection can be re-attempted within the `logger.writeLoop()` function.